src/Plugin/DataType/StrawberryValuesViaJmesPathFromJson.php

```diff
@@ -93,12 +98,21 @@ public function process($langcode = NULL)
         foreach ($values as $value) {
           $result = $parser->parse($value);
           if ($result->isValid()) {
-            $date_ini = $result->getEdtfValue()->getMin();
-            $date_max = $result->getEdtfValue()->getMax();
-            $date_ini = date('c', $date_ini);
-            $date_max = date('c', $date_max);
-            $values_parsed[] = $date_ini;
-            $values_parsed[] = $date_max;
+            $edtf_value = $result->getEdtfValue();
+            switch(get_class($edtf_value)) {
+              case "EDTF\Model\Interval":
```

Comment:
Ok, I do not get this part. And if it is not an interval?
you still do
$values_parsed[] = date('c', $edtf_value->getMin());
$values_parsed[] = date('c', $edtf_value->getMax());
?

A bit confused, could you explain?

Reply:
This could just be an if/else, but when I wrote it I wasn't sure what other classes that $result->getEdtfValue() might return, so I did it as a switch. Default is to do it as before, but if the resulting edtf date is an interval, then check for presence of start and end dates before trying to getMin/Max.

Reply:
Ok. That makes sense, So again. This is a bug in `EDTF EDTF\Model\Interval` because as explained in the issue the constructor of that class does not allow an empty/null start nor end... so it should always have a return... interface also applies to every type of edtfValue. Did I get that right?

Reply:
yes I think you get it